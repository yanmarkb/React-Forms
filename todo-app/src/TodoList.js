import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid";

const TodoList = () => {
	const [todos, setTodos] = useState([]);

	const addTodo = (newTodo) => {
		setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
	};

	const removeTodo = (id) => {
		setTodos((todos) => todos.filter((todo) => todo.id !== id));
	};

	return (
		<div>
			<NewTodoForm addTodo={addTodo} />
			<div>
				{todos.map((todo) => (
					<Todo
						key={todo.id}
						id={todo.id}
						task={todo.task}
						removeTodo={removeTodo}
					/>
				))}
			</div>
		</div>
	);
};

export default TodoList;
