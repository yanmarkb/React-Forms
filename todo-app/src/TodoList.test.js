import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList";

test("TodoList component renders without crashing", () => {
	render(<TodoList />);
});

test("TodoList component matches snapshot", () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});
