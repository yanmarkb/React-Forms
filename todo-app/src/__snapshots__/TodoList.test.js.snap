// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
